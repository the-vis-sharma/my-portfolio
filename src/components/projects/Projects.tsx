import { useState } from "react";
import {
  Box,
  Button,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Grid,
  Typography,
} from "@mui/material";
import projects from "./data";
import ProjectDetail from "./ProjectDetail";
import { Project } from "./project.model";

const Projects = () => {
  const [openProjectDetail, setOpenProjectDetail] = useState(false);
  const [selectedProject, setSelectedProject] = useState(projects[0]);

  const handleCardClick = (project: Project) => {
    setSelectedProject(project);
    setOpenProjectDetail(true);
  };
  return (
    <Box
      id="projects-section"
      sx={{
        p: 4,
        pt: 8
      }}
    >
      <Typography variant="h3" textAlign="center" gutterBottom color="primary" fontWeight="500">
        Projects
      </Typography>

      <Grid container spacing={2} justifyContent="center" alignItems="center">
        {projects.map((project) => (
          <Grid
            item
            key={project.name}
            lg={4}
            md={6}
            sm={12}
            sx={{ position: "relative" }}
          >
            <Card
              onClick={() => handleCardClick(project)}
              data-testid="project-card"
            >
              <CardMedia
                component="img"
                alt="green iguana"
                height="140"
                image={project.image}
              />
              <CardContent>
                <Typography
                  gutterBottom
                  variant="h5"
                  component="div"
                  data-testid="project-title"
                  color="primary"
                >
                  {project.name}
                </Typography>
                <Typography variant="body2" color="text.secondary">
                  {project.description}
                </Typography>
              </CardContent>
              <CardActions>
                <Button size="small">Learn More</Button>
              </CardActions>
            </Card>
          </Grid>
        ))}
      </Grid>
      <ProjectDetail
        project={selectedProject}
        open={openProjectDetail}
        handleClose={() => setOpenProjectDetail(false)}
      />
    </Box>
  );
};

export default Projects;
