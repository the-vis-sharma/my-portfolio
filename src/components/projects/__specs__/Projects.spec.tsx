import { act, fireEvent, render } from "@testing-library/react";
import Projects from "../Projects";

describe("Projects", () => {
  it("verify snapshot for projects", () => {
    const { asFragment } = render(<Projects />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("project detail modal should open on click of project card", () => {
    const { getAllByTestId, getByTestId } = render(<Projects />);
    const projectCard = getAllByTestId("project-card");
    act(() => {
      fireEvent.click(projectCard[2]);
    });

    expect(getByTestId("modal-project-title").textContent).toEqual(
      getAllByTestId("project-title")[2].textContent
    );
  });
});
