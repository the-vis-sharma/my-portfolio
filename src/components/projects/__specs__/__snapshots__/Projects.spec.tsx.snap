// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify snapshot for projects 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1wcaknn"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-lxmg88-MuiTypography-root"
    >
      Projects
    </h3>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-6s1xft-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              Paisa Vasool
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              It's a expense management and tracker app which also let you split the expenses among your friends and groups.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              U-Move: Bike Rental App
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              A Bike Rental Project with Angular 8 Admin portal, Ionic 4 App for android & iOS with spring boot for backend.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              Smart Queuing System
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              This is used to detect the people in queue to smartly manage the queue at various places in our day to day life.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              People Counter App
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              The app will detect people in a designated area, providing the number of people in the frame, the average duration of people spend and total count.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              Status Downloader for WhatsApp
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              This is a android application to download and share WhatsApp status into your.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              DP & Videos Downloader
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              View & Download Instagram Profile Picture & Videos in full & HD Resolution and Zoom Picture with The DP & Videos Downloader app.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              Red Basket - Groceries Shopping App
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              This is a clone of Big Basket App.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <img
            alt="green iguana"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test-file-stub"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              data-testid="project-title"
            >
              Fuel Trades
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              Web app that helps Boston teens find the nearest available short-term housing.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
